src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -260,6 +261,11 @@ public XmlSerializer(Type type, string defaultNamespace)
             _mapping = GetKnownMapping(type, defaultNamespace);
             if (_mapping != null)
             {
+                if (Mode == SerializationMode.PreGenOnly)
+                {
+                    throw new NotSupportedException("Not support to use PreGenOnly mode since there's known mapping.");
```

Comment:
Do we need to throw exception from here? You already throw exception when LoadGeneratedAssembly returns null.